### PR TITLE
Make the original ESP_IDF_VERSION available, fix event loop legacy and WiFi struct on v5.0+

### PR DESCRIPTION
**Pull request by adokitkat**
I tried to compile the default example from README on IDF v5.0+ but it didn't work so this should fix it.

I tested it on IDF v5.0 and v5.2 (with e.g. `nimble esp_build --esp-idf-version=V5.2`) and it worked.

I also renamed ESP_IDF_VERSION to ESP_IDF_VERSION_STR so I can implement real ESP_IDF_VERSION from IDF, hope it isn't a problem.

### CODE REVIEW

File: src/nesper/consts.nim [hunk 1]
```diff
@@ -10,17 +10,27 @@ import strutils
 from os import getEnv
 from macros import error, warning
 
+template ESP_IDF_VERSION_VAL*(major, minor, patch: int): int = (major shl 16 or minor shl 8 or patch)
+
 const
-  ESP_IDF_VERSION* {.strdefine.} = getEnv("ESP_IDF_VERSION", "0.0").toLower().replace("v","")
-  ESP_IDF_MAJOR* {.intdefine.} = ESP_IDF_VERSION. split(".")[0].parseInt()
-  ESP_IDF_MINOR* {.intdefine.} = ESP_IDF_VERSION.split(".")[1].parseInt()
+  ESP_IDF_VERSION_STR* {.strdefine.} = getEnv("ESP_IDF_VERSION_STR", "0.0.0").toLower().replace("v","")
```
Comment (elcritch) [inline src/nesper/consts.nim]: I wanted to be able to re-use the `ESP_IDF_VERSION` env variable that gets set. Did you find it clashed with things?

Comment (adokitkat) [reply src/nesper/consts.nim]: Sorry for the late reply. I see, I will look into it. I guess we can do `getEnv("ESP_IDF_VERSION", "0.0")`, but `ESP_IDF_VERSION` in code should be same as in the IDF in my opinion se we can do eqivalent of e.g. `#if ESP_IDF_VERSION >= ESP_IDF_VERSION_VAL(5, 0, 0)`.

Comment (adokitkat) [reply src/nesper/consts.nim]: How about now?

Comment (elcritch) [reply src/nesper/consts.nim]: > Sorry for the late reply. I see, I will look into it. I guess we can do getEnv("ESP_IDF_VERSION", "0.0"), but ESP_IDF_VERSION in code should be same as in the IDF in my opinion se we can do eqivalent of e.g. #if ESP_IDF_VERSION >= ESP_IDF_VERSION_VAL(5, 0, 0)

I agree that `ESP_IDF_VERSION >= ESP_IDF_VERSION_VAL(5, 0, 0)` is better. Why do we need the `ESP_IDF_VERSION_STR` version? Couldn't we just swap the when checks to use `ESP_IDF_VERSION_VAL`?

Comment (adokitkat) [reply src/nesper/consts.nim]: Now `ESP_IDF_VERSION` environment variable can be used as well as `-d:ESP_IDF_VERSION=` the same but it's bound to `ESP_IDF_VER` in Nim code and `ESP_IDF_VERSION` in Nim code is the correct one.

I left `ESP_IDF_VERSION_STR` there as a template which just stringifies `ESP_IDF_MAJOR` etc.

**Pull request merged**